torch_geometric/data/hetero_data.py

```diff
@@ -667,6 +674,24 @@ def __len__(self) -> int:
     def __iter__(self):
         raise NotImplementedError
 
+    # MaterializedGraph interface #############################################
+
+    def _layout_to_attr_name(self, layout: EdgeLayout) -> str:
```

Comment:
can we make this part of `MaterializedGraph`

Reply:
I don't think we want this, since `_layout_to_attr_name` is specific to PyG (and will likely not generalize to any other backing feature stores).